src/components/Modals/MenuProductModal.tsx

```diff
@@ -0,0 +1,136 @@
+import { ReactElement, useCallback, useState } from 'react';
+import { ProductDto } from 'dtos/ProductDto';
+import { Button } from 'components/Button/Button';
+import { useLocalOrder } from 'context/OrderContext';
+import PlaceHolderImage from 'assets/burger_transparent.png';
+import { ModalTransition } from 'components/Transitions/ModalTransition';
+import { motion } from 'framer-motion';
+import { GBPFormat } from 'utilities/GBPFormat';
+
+interface Props {
+  product: ProductDto;
+  showProductModal: boolean;
+  disableShow: (isShow: boolean) => void;
+}
+
+export const MenuProductModal = ({ product, showProductModal, disableShow }: Props): ReactElement => {
+  const { addToLocalOrder, removeProduct, localOrder } = useLocalOrder();
+
+  const checkQuantity = (product: ProductDto): number => {
+    const existingItem = localOrder.lineItems.find((item) => item.product.id === product.id);
+    return existingItem ? existingItem.quantity : 1;
+  };
+
+  const [counter, setCounter] = useState(checkQuantity(product));
+
+  const changeCount = (action: number): void => {
+    if (counter + action >= 0) {
+      setCounter(counter + action);
+    }
+  };
+
+  const addToOrder = useCallback(() => {
+    addToLocalOrder(product, counter);
+  }, [addToLocalOrder, product, counter]);
+
+  const allergens = ['Milk', 'Peanuts', 'Celery', 'Soy', 'Gluten'];
+
+  return (
+    <>
+      {showProductModal && (
+        <div
+          className="fixed top-0 left-0 z-10 flex h-full w-full items-center justify-center bg-black bg-opacity-75 shadow-sm"
+          onClick={() => disableShow(false)}
+        >
+          <ModalTransition>
+            <div
+              className="flex w-[70%] flex-col items-center rounded-2xl bg-white p-[20px] text-[#333F4C] shadow-[0_4px_4px_rgba(0,0,0,0.25)]"
+              onClick={(e) => e.stopPropagation()}
+            >
+              <div className="flex w-full flex-row justify-start gap-[20px] ">
+                <img alt="burger" className="h-[18vh] w-[100%] rounded-[8px] object-cover" src={PlaceHolderImage} />
+                <div className="flex w-full flex-col items-start justify-between gap-2">
+                  <h5 className="text-[20px] font-[600]">{product.name}</h5>
+                  <p className="font-poly text-left italic">{product.description}</p>
+                  <div className="flex w-full items-center justify-between gap-2">
+                    <div className="font-[600]">V, Ve, GF</div>
+                    <div className="flex items-center">
+                      <motion.button
```

Comment:
This needs to be refactored into it's own component. Wasn't sure straight away the best way to go about this, but will have a look soon